app/views/Main.js

```diff
@@ -19,18 +19,30 @@ import PermissionsContext, {
 import { useSelector } from 'react-redux';
 import selectedHealthcareAuthoritiesSelector from '../store/selectors/selectedHealthcareAuthoritiesSelector';
 import { useStatusBarEffect } from '../navigation';
+import { useDeepCompareEffect } from '../helpers/CustomHooks';
+import { debounce } from 'lodash';
 
 export const Main = () => {
   useStatusBarEffect('light-content');
   const navigation = useNavigation();
   const { notification } = useContext(PermissionsContext);
   const hasSelectedAuthorities =
     useSelector(selectedHealthcareAuthoritiesSelector).length > 0;
+
+  const debounceUpdateStateInfo = useRef(
```

Comment:
Won't this delay the UI state changes by a second? Should this be throttled?

Reply:
Debouncing is removed because that is now handled in a similar way in the `IntersectService` class.